zebedee-cms/src/test/java/com/github/onsdigital/zebedee/keyring/LegacyKeyringImplTest.java

```diff
@@ -569,70 +617,113 @@ public void testAdd_collectionDescriptionNull_shouldThrowException() {
+        KeyringException actual = assertThrows(KeyringException.class, () -> legacyKeyring.add(user, collection, null));
 
         // Then an exception is thrown
-        assertThat(ex.getMessage(), equalTo(COLLECTION_ID_EMPTY_ERR));
+        assertThat(actual.getMessage(), equalTo(COLLECTION_ID_EMPTY_ERR));
+        verifyZeroInteractions(keyringCache, users, userKeyring, cachedUserKeyring);
     }
 
     @Test
-    public void testAdd_collectionIDEmpty_shouldThrowException() {
+    public void testAdd_collectionDescriptionIDEmpty_shouldThrowException() {
         // Given collection ID is empty
         when(collectionDescription.getId())
                 .thenReturn("");
 
         // When add is called
-        KeyringException ex = assertThrows(KeyringException.class, ()-> keyring.add(user, collection, null));
+        KeyringException actual = assertThrows(KeyringException.class, () -> legacyKeyring.add(user, collection, null));
 
         // Then an exception is thrown
-        assertThat(ex.getMessage(), equalTo(COLLECTION_ID_EMPTY_ERR));
+        assertThat(actual.getMessage(), equalTo(COLLECTION_ID_EMPTY_ERR));
+        verifyZeroInteractions(keyringCache, users, userKeyring, cachedUserKeyring);
     }
 
     @Test
-    public void testAdd_secretKeyNull_shouldThrowException() {
-        // Given secret key is null
+    public void testAdd_keyNull_shouldThrowException() {
+        // Given key is null
 
         // When add is called
-        KeyringException ex = assertThrows(KeyringException.class, ()-> keyring.add(user, collection, null));
+        KeyringException actual = assertThrows(KeyringException.class, () -> legacyKeyring.add(user, collection, null));
 
         // Then an exception is thrown
-        assertThat(ex.getMessage(), equalTo(SECRET_KEY_NULL_ERR));
+        assertThat(actual.getMessage(), equalTo(SECRET_KEY_NULL_ERR));
+        verifyZeroInteractions(keyringCache, users, userKeyring, cachedUserKeyring);
+    }
+
+    @Test
+    public void testAdd_cacheKeyringNotFound_shouldUpdatedStoredUserOnly() throws Exception {
+        // Given a cached keyring does not exist for the user
+        when(keyringCache.get(user))
+                .thenReturn(null);
+
+        // When add is called
+        legacyKeyring.add(user, collection, secretKey);
+
+        // Then nothing happens
+        verify(keyringCache, times(1)).get(user);
+        verify(users, times(1)).addKeyToKeyring(TEST_EMAIL, TEST_COLLECTION_ID, secretKey);
+        verifyZeroInteractions(userKeyring, cachedUserKeyring);
+    }
+
+    @Test
+    public void testAdd_cacheKeyringIsLocked_shouldUpdateStoredUserOnly() throws Exception {
+        // Given the users cached keyring is locked
+        when(cachedUserKeyring.isUnlocked())
+                .thenReturn(false);
+
+        // When add is called
+        legacyKeyring.add(user, collection, secretKey);
+
+        // Then the keyring cache is not updated
+        verify(keyringCache, times(1)).get(user);
```

Comment:
I think there should be another verify here to check that the keyring cache is not updated?